### PR TITLE
Add `source_stmt` in `Context` for test logging.

### PR DESCRIPTION
**Pull request by ekexium**
ref tikv/tikv#14474

### CODE REVIEW

File: proto/kvrpcpb.proto
Comment (cfzjywxk) [inline proto/kvrpcpb.proto:923]: How about `QueryCtx` or `StmtCtx`? Our goal is to establish a connection between the execution flow of transactions or statements.
`connection_id` may not be globally unique, to clarify which query the request belongs to something like `stmt_id` or `uuid` is needed but it looks a bit complex.

File: proto/kvrpcpb.proto
Comment (zyguan) [inline proto/kvrpcpb.proto:868]: If this field is only used for logging with context info, I think a formated string is enough (just like `request_source`).

Comment (ekexium) [reply proto/kvrpcpb.proto:868]: IMO string is less structured and less flexible.

**Comment by disksing** [comment]
@cfzjywxk @zyguan Please take another look.

File: proto/kvrpcpb.proto
Comment (cfzjywxk) [inline proto/kvrpcpb.proto:949]: Since the `connection_id` can be identical across different TiDB servers, it may be necessary to use a unique identifier such as `uuid` to indicate which statement and instance this query originates from.
Besides, how should the `stmt_id` be generated?

**Review by zyguan — APPROVED** [review]
LGTM, is the source_stmt field only set for testing?

BTW, source_stmt.start_ts seems to be a little redundant since most of rpc requests already have a field that indicate to start-ts.

File: proto/kvrpcpb.proto
Comment (ekexium) [reply proto/kvrpcpb.proto:949]: > how should the stmt_id be generated

Each connection generates its own stmt_id. 
Merging them together into a UUID can make it harder for humans to read. Since it's only used in tests, the performance gain doesn't matter a lot.

**Comment by ekexium** [comment]
We can wait until the corresponding TiKV PR completes its review before merging this PR

**Pull request merged**
